### PR TITLE
[Type] Update AnalysisResult

### PR DESCRIPTION
**Pull request by takesei**
* TaguPy version:
* Python version:
* Operating System:

### What
Update `tagupy.type.AnalysisResult` as follows
- Define `_AnalysisResult` as ABC
- Define `AnalysisResult` as alias of Type[_AnalysisResult]

When you make a new stats result as a subclass of AnalysisResult, use `_AnalysisResult` as a superclass.
When you make a function that requires/returns a kind of AnalysisResult, use `AnalysisResult`

### CODE REVIEW

File: tagupy/type/_analysis_result.py [hunk 1]
```diff
@@ -1,12 +1,34 @@
 """
 NamedTuple based Data Structure
```
Comment (taka-sho) [inline tagupy/type/_analysis_result.py:2]: こちら `NameTuple` を使わなくなったので，他の表現の方が適切なのかなと思いました

Comment (takesei) [reply tagupy/type/_analysis_result.py]: fix done, close

**Review by gk-hazard — COMMENTED** [review]
lgtm

**Review by gk-hazard — APPROVED** [review]
lgtm

**Comment by gk-hazard** [comment]
> lgtm

ごめんミスった

**Pull request merged**